src/v/cluster/members_backend.cc

```diff
@@ -308,6 +308,28 @@ ss::future<> members_backend::reconcile() {
     if (!_raft0->is_leader() || _updates.empty()) {
         co_return;
     }
+
+    // use linearizable barrier to make sure leader is up to date and all
+    // changes are applied
+    auto barrier_result = co_await _raft0->linearizable_barrier();
+    if (
+      barrier_result.has_error()
+      || barrier_result.value() < _raft0->dirty_offset()) {
+        if (!barrier_result) {
+            vlog(
+              clusterlog.debug,
+              "error waiting for all raft0 updates to be applied - {}",
+              barrier_result.error().message());
+        }
+        vlog(
+          clusterlog.trace,
+          "waiting for all raft0 updates to be applied - barrier offset: {}, "
+          "raft0 dirty offset: {}",
+          barrier_result.value(),
+          _raft0->dirty_offset());
+        co_return;
+    }
```

Comment:
can we clean-up this condition so the error is handled in a separate block from the non-error case? beyond that does this have undefined behavior? in the error case after the `barrier_result.error().message()` is logged we drop through and look at the `barrier_result.value()` value, too.

Reply:
great catch.